File: _includes/sort-benchmark/helpers/partition_at.rs
fn partition_at(a: &mut [usize], lo: usize, hi: usize, pivot_idx: usize) -> usize {
    a.swap(pivot_idx, hi);
    let pivot = a[hi];
    let mut i = lo;
    for j in lo..hi {
        if a[j] < pivot {
            a.swap(i, j);
            i += 1;
        }
    }
    a.swap(i, hi);
    i
}

File: _includes/sort-benchmark/helpers/partition_at.swift
func partition_at(_ a: UnsafeMutableBufferPointer<Int>, _ lo: Int, _ hi: Int, _ pivot_idx: Int) -> Int {
    a.swapAt(pivot_idx, hi)
    let pivot = a[hi]
    var i = lo
    for j in lo..<hi {
        if a[j] < pivot {
            a.swapAt(i, j)
            i += 1
        }
    }
    a.swapAt(i, hi)
    return i
}
